### PR TITLE
HDDS-9611. Avoid unnecessary buffer conversion.

### PR DESCRIPTION
**Pull request by szetszwo**
## What changes were proposed in this pull request?

In KeyValueHandler.validateChunkChecksumData(..), it converts ChunkBuffer to ByteString, which will be converted back to ChunkBuffer.

Also includes code cleanup:
- avoid passing buffer capacity as `long`;
- check for integer overflow

All the changes were discovered in HDDS-9536.

## What is the link to the Apache JIRA

HDDS-9611

## How was this patch tested?

Updating existing tests.

### CODE REVIEW

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/ChunkUtils.java [hunk 1]
```diff
@@ -180,6 +183,15 @@ private static long writeDataToChannel(FileChannel channel, ChunkBuffer data,
     }
   }
 
+  public static ChunkBuffer readData(long len, int bufferCapacity,
+      CheckedConsumer<ByteBuffer[], StorageContainerException> readMethod)
+      throws StorageContainerException {
+    final ByteBuffer[] buffers = BufferUtils.assignByteBuffers(len,
+        bufferCapacity);
+    readMethod.accept(buffers);
+    return ChunkBuffer.wrap(Lists.newArrayList(buffers));
```
Comment (adoroszlai) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/helpers/ChunkUtils.java:192]: `ChunkBuffer` doesn't change the list's size, and it creates a defensive copy anyway.  So I think we could use `Arrays.asList()` here.

**Review by adoroszlai — COMMENTED** [review]
Thanks @szetszwo for the patch.  LGTM, but there are some conflicts due to recent commit e4ff5fa1b9b6a8335e5e1d7a118c9c3d4b46281d.

**Comment by szetszwo** [comment]
@adoroszlai , Thanks for reviewing this!  Sync'ed with the master branch and using `Arrays.asList()` now.

**Review by adoroszlai — APPROVED** [review]

**Comment by szetszwo** [comment]
@adoroszlai , thanks for merging this!

**Pull request merged**